Minor fix: remove redundant tag name in error message of create failed.

coolljt0725 opened the pull request:
Signed-off-by: Lei Jitang leijitang@huawei.com
When user rest API to create a container with a image which is 
not exist on local, the error message has a redundant tag.

<pre><code>$ curl -X POST http://9.81.1.183:2375/containers/create -H "Content-Type: application/json" -d '{
"Image": "test:v3"
}'
No such image: test:v3 (tag: v3)</code></pre>

and if we use a image use default tag, the error message seems good

<pre><code> curl -X POST http://9.81.1.183:2375/containers/create -H "Content-Type: application/json" -d '{
"Image": "test"
}'
No such image: test (tag: latest)</code></pre>

calavera commented:
do you mind to add regression tests?

coolljt0725 commented:
@calavera  updated, add a test

calavera commented:
LGTM, thanks for adding the test, but it needs a rebase :pensive:

coolljt0725 commented:
@calavera  rebased.  and change return message from `"No such image: %s (tag: %s)"`
to `"No such image: %s:%s"` due to the client side https://github.com/docker/docker/blob/master/api/client/create.go#L117
will judge if the return message has `config.Image`. The `config.Image`
could be the format `repo` or `repo:tag`, so we should return `"%s:%s", img, tag`.
sorry you have to review this again. :-)

calavera commented:
I think you have to update your test too, they still expect the old format: `"No such image: test456 (tag: latest)"`

coolljt0725 commented:
@calavera  thanks you, updated. Sorry,  I have updated on local, working with mutiple terminals made me push the branch on another machine to the github.....

cpuguy83 commented:
Looks like all the tests with images references by digest are broken.

coolljt0725 commented:
@cpuguy83  test failed fixed, the reason is that docker use `@` to divide repo and tag when use digest.

cpuguy83 commented:
@coolljt0725 Could we keep the same pattern as the old error but instead of using `config.Image` have it use the parsed image name?

coolljt0725 committed:
Minor fix: remove redundant tag name in error message of create failed.

Signed-off-by: Lei Jitang <leijitang@huawei.com>

coolljt0725 commented:
@cpuguy83  sorry for the late response. updated, is this what you want?

cpuguy83 commented:
@coolljt0725 I'm not sure what I was thinking :) I think I was hoping the parsed image + tag would include the `@` and therefore not need a special case...
In any case...
LGTM
